ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -577,6 +577,86 @@ public void deleteErrorCasesTest() {
     verifyDeleteFailure(getUniqueId(), StoreErrorCodes.ID_Not_Found);
   }
 
+  /**
+   * Test DELETE authorization success for {@link BlobStore#delete(MessageWriteSet)}.
+   * If either accountId or containerId is -1, validation is skipped.
+   */
+  @Test
+  public void deleteAuthorizationSuccessTest() throws Exception {
+    short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1};
+    short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM)};
+    for (int i = 0; i < accountIds.length; i++) {
+      MockId mockId = putSingleBlob(UtilsTest.getRandomString(10), accountIds[i], containerIds[i]);
+      MessageInfo info = new MessageInfo(mockId, DELETE_RECORD_SIZE, Utils.getRandomShort(TestUtils.RANDOM),
+          Utils.getRandomShort(TestUtils.RANDOM), System.currentTimeMillis());
+      MessageWriteSet writeSet = new MockMessageWriteSet(Collections.singletonList(info),
+          Collections.singletonList(ByteBuffer.allocate(DELETE_RECORD_SIZE)));
+      store.delete(writeSet);
```

Comment:
you aren't changing the account and container ID in the ID though, should you be doing that to make this test relevant (like you do in the equivalent test in GET)?

Reply:
You mean try to do store.delete() with different combination of accountId/containerId, right?